From 414f25fb42d7a577ece9eda0ef429eb2bf378107 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Sat, 5 Nov 2022 21:54:37 +0200
Subject: [PATCH 08/11] [openmp] [test] Set __COMPAT_LAYER=RunAsInvoker when
 running tests on Windows

Windows heuristics may decide to want to run some tested processes
as elevated (since it may think some of them are installers - executables
with "dispatch" in the name may hit a heuristic looking for "patch").

Set this environment variable to disable this heuristic and just run
the executable with whatever privileges the caller has.

This fixes a couple tests on such versions of Windows where this
heuristic is active.

Differential Revision: https://reviews.llvm.org/D137772
---
 openmp/runtime/test/lit.cfg | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/openmp/runtime/test/lit.cfg b/openmp/runtime/test/lit.cfg
index a8dfdff29b2e..491fbdbf445f 100644
--- a/openmp/runtime/test/lit.cfg
+++ b/openmp/runtime/test/lit.cfg
@@ -108,6 +108,9 @@ if config.operating_system == 'NetBSD':
 if config.operating_system in ['Linux', 'Windows']:
     config.available_features.add('affinity')
 
+if config.operating_system == 'Windows':
+    config.environment['__COMPAT_LAYER'] = 'RunAsInvoker'
+
 import multiprocessing
 try:
     if multiprocessing.cpu_count() > 1:
-- 
2.25.1

